app/web/features/search/FilterDialog.tsx

```diff
@@ -85,17 +85,17 @@ interface FilterModalFormData
 interface FilterDialogProps {
   isOpen: boolean;
   onClose(): void;
-  queryName: undefined | string;
-  setQueryName: Dispatch<SetStateAction<undefined | string>>;
-  setLocationResult: any;
+  queryName: string;
```

Comment:
I get confused here, when there is no query I want to send undefined to the server... should be allowed to have undefined type then?

Reply:
True, I could use an empty string and then if empty string then send an undefined to the server... that's also right :) [I'm my case I'm more used to undefined = something not set, and "" as empty string (but actual value)

Reply:
actually you are right... it's more stable having 1 type, that 2, makes code more "robust" in my opinion... keeping your change!